src/sync/clients.ts

```diff
@@ -92,3 +95,55 @@ export async function setClient(
   clients.set(id, client);
   return setClients(clients, dagWrite);
 }
+
+export async function initClient(
+  dagStore: dag.Store,
+): Promise<[ClientID, Client]> {
+  return dagStore.withWrite(async dagWrite => {
+    const clients = await getClients(dagWrite);
+    const newClientID = makeUuid();
+    const bootstrapClient = [...clients.values()].reduce((prev, curr) =>
```

Comment:
this is not hot code but you could have done this with a loop so you do not have to allocate that temp array.

Reply:
I'm going to just write lazy map/reduce/filter for iterator because IMHO functional codes intent is quicker to grok then declarative.

Reply:
went ahead and made this a for/of until I manage to write some functional iterator helpers